projects/aca-folder-rules/src/lib/rule-details/actions/rule-action.ui-component.ts

```diff
@@ -82,10 +94,14 @@ export class RuleActionUiComponent implements ControlValueAccessor, OnInit, OnDe
     return this.actionDefinitions.find((actionDefinition: ActionDefinitionTransformed) => actionDefinition.id === this.selectedActionDefinitionId);
   }
 
+  get cardViewStyle() {
+    return this.isFullWidth ? { ' width': '100%' } : {};
+  }
+
   onChange: (action: RuleAction) => void = () => undefined;
   onTouch: () => void = () => undefined;
 
-  constructor(private cardViewUpdateService: CardViewUpdateService) {}
+  constructor(private cardViewUpdateService: CardViewUpdateService, private folderRulesService: FolderRulesService) {}
```

Comment:
This is a UI component, meaning it should not have any business logic in it. `CardViewUpdateService` is acceptable because it's a service which interacts directly with the card view component but the `FolderRuleService` calls the API. Can you do something similar to what I did with the `actionDefinitions`?

Reply:
To be clear, ideally we'd want an input property which takes a list of aspects, which could originate from anywhere, not just the API. It makes the component more reusable. The `edit-rule-dialog` and `manage-rules` components are, on the other hand, smart component which can use these services.

Reply:
Should `rule-action.ui-component.ts` receive already parsed aspects or a raw list. In other words I am not sure if `parseAspectsToSelectOptions` method can be a part of the `rule-action.ui-component.ts` component?

Reply:
`parseAspectsToSelectOptions` sounds to me like component logic. If it's something only the component needs for a select dropdown contained within, I think it's good where it is.